src/Data/Text/Internal/Lazy/Search.hs

```diff
@@ -122,6 +122,13 @@ indicesOne c = chunk
              | otherwise = go (h+1)
              where on = A.unsafeIndex oarr (ooff+h)
 
+-- | First argument is a strict Text, and second is a lazy one.
+isPrefixOf :: T.Text -> Text -> Bool
```

Comment:
Could maybe also add a test for this just for safety?

Reply:
It would be awkward to export this helper. It is covered by tests for `indices` and I just added one more for truly pathological chunks.